src/main/kotlin/de/bund/digitalservice/useid/config/ContentSecurityPolicyFilter.kt

```diff
@@ -0,0 +1,49 @@
+package de.bund.digitalservice.useid.config
+
+import de.bund.digitalservice.useid.widget.INCOMPATIBLE_PAGE_PATH
+import de.bund.digitalservice.useid.widget.WIDGET_PAGE_PATH
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.http.server.reactive.ServerHttpResponse
+import org.springframework.web.server.ServerWebExchange
+import org.springframework.web.server.WebFilter
+import org.springframework.web.server.WebFilterChain
+import org.springframework.web.util.pattern.PathPattern
+import org.springframework.web.util.pattern.PathPatternParser
+import reactor.core.publisher.Mono
+
+class ContentSecurityPolicyFilter(
+    private val contentSecurityPolicyProperties: ContentSecurityPolicyProperties
+) : WebFilter {
+
+    private val widgetPagePath: PathPattern = PathPatternParser().parse(WIDGET_PAGE_PATH)
+    private val incompatiblePagePath: PathPattern = PathPatternParser().parse(INCOMPATIBLE_PAGE_PATH)
+    private val listOfPages = listOf(widgetPagePath, incompatiblePagePath)
+
+    override fun filter(
+        serverWebExchange: ServerWebExchange,
+        webFilterChain: WebFilterChain
+    ): Mono<Void> {
+        val request: ServerHttpRequest = serverWebExchange.request
+        val response: ServerHttpResponse = serverWebExchange.response
+        val pathIsValidWidgetPages = listOfPages.any { it.matches(request.path.pathWithinApplication()) }
+
+        if (pathIsValidWidgetPages) {
```

Comment:
@mpanne @hannesschaletzky I am not happy with the nested if statement, do you know how can we write this cleaner?

Reply:
You could just invert it: 

```
if (!pathIsValidWidgetPages) {
    return webFilterChain.filter(serverWebExchange)
}
```